### PR TITLE
:sparkles: Fetch workspace data in batches and add advanced filtering

### PR DESCRIPTION
**Pull request by foysalit**
Depends on https://github.com/bluesky-social/atproto/pull/2836

Use the newly available getRepos and getRecords endpoints to bulk fetch data for workspace items. This also allows us to provide additional context to workspace items without expanding into full view mode per item.

Screenshot 2024-09-30 at 17 09 32

This PR also adds a way to filter items from the workspace based on a combination of options among accounts and other content type which allows mods to select/unselect items from workspace in batches.

Screenshot 2024-10-04 at 20 46 33

### CODE REVIEW

File: components/common/labels/List.tsx [hunk 1]
```diff
@@ -106,7 +106,7 @@ export const ModerationLabel = ({
   })
 
   return (
-    <Popover className="relative">
+    <Popover className="relative z-20">
```
Comment (foysalit) [inline components/common/labels/List.tsx:109]: this known z-index value helps us overlap other content on labels.

File: components/workspace/Panel.tsx [hunk 2]
```diff
@@ -35,23 +36,13 @@ export function WorkspacePanel(props: PropsOf<typeof ActionPanel>) {
...
-  }
-
   const handleRemoveSelected = () => {
     const selectedItems = Array.from(
       formRef.current?.querySelectorAll<HTMLInputElement>(
         'input[type="checkbox"][name="workspaceItem"]:checked',
       ) || [],
     ).map((checkbox) => checkbox.value)
-    removeItemsMutation.mutate(selectedItems)
+    removeItemsMutation.mutate(selectedItems as string[])
```
Comment (devinivy) [inline components/workspace/Panel.tsx:45]: Are the checkbox values always strings, never e.g. `undefined`?

**Review by devinivy — APPROVED** [review]
Nice 👍

File: components/workspace/Panel.tsx (hunk 2 above)
Comment (foysalit) [reply components/workspace/Panel.tsx:45]: yep, these checkboxes always carry the value of the did/uri

**Pull request merged**